compiler/src/iree/compiler/GlobalOptimization/PropagateLinalgTranspose.cpp

```diff
@@ -735,6 +736,22 @@ class BubbleTransposeThroughUnaryElementwiseDpsInit
   }
 };
 
+// Folds `linalg.transpose(tensor.empty(), dest)` to `dest`.
+class FoldTransposeOfEmpty : public OpRewritePattern<linalg::TransposeOp> {
```

Comment:
I'm surprised that we need the pattern. How does it happen?

Reply:
It's because one of the patterns to propagate transposes introduces a transpose on the destination. I could make that pattern smarter and check if the destination is an empty.

Reply:
Are they all introduced by `createTranspose` op? If so, I think we can rename it to `createTransposeOrEmptyOp` and add the doc to it. What do you think?

Reply:
oh yeah that's a good idea, let me try that